tests/bot/bot_test.py

```diff
@@ -18,10 +18,26 @@ def test_send_dm(self):
 
         bot.send_dm("Hahahaha", "UD8UCTN05")
         self.mock_sc.api_call.assert_called_with(
-                                        'chat.postMessage',
-                                        text="Hahahaha",
-                                        channel="UD8UCTN05"
-                                        )
+            'chat.postMessage',
+            text="Hahahaha",
+            channel="UD8UCTN05"
+        )
+
+    def test_send_dm_failure(self):
+        """Test send_dm() when the Slack API call fails."""
+        bot = Bot(self.mock_sc)
+        self.mock_sc.api_call = mock.MagicMock(return_value={"error": "Error"})
+
+        try:
+            bot.send_dm("Hahahaha", "UD8UCTN05")
+        except SlackAPIError as e:
+            assert e.error == "Error"
+        finally:
```

Comment:
First time I've ever seen anyone use `finally`!

Reply:
Yeah, not sure it’s actually useful here... but it does signal intent, so 🤷‍♀️